include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -553,26 +573,35 @@ class argument_parser
             return;
         }
 
+        bool special_format_was_set{false};
+        bool sub_parser_was_set{false};
+
         for(int i = 1, argv_len = argc; i < argv_len; ++i) // start at 1 to skip binary name
         {
             std::string arg{argv[i]};
 
+            if (std::ranges::find(subcommands, arg) != subcommands.end())
+            {
+                sub_parser = std::make_unique<argument_parser>(info.app_name + "-" + arg, argc - i, argv + i, false);
+                sub_parser_was_set = true;
```

Comment:
I think this won't work in the following case:

```
git --string-option pull --flag bar push --foo
```
In this case it will identify the subcommand as pull, although it is push, or not?

Reply:
Why don't you call parse on the the sub_parser here and return after that? Then you don't need `special_format_was_set` and the extra handling at the end.

Reply:
> I think this won't work in the following case:

Yes you are right but I do not see how I can avoid that?

> Why don't you call parse on the the sub_parser here and return after that?

Because the sub parser is not set up yet by the user with all its `add_option` calls yet?

Reply:
* forbid options and positionals in the top-level parser if it has subcommends
* forbid subcommands that start with `-` or consist only of numbers

Reply:
If we do not allow any options anyway we don't need to restrict the command to "may not have only numbers"?

Reply:
I don't core.